.github/workflows/main.yml

```diff
@@ -111,8 +111,12 @@ jobs:
                   workspace=$(jq -r '.name' packages/${{ matrix.dir }}/package.json)
                   yarn workspace "$workspace" run test:coverage
 
-            - if: github.event_name == 'push' && github.ref == 'refs/heads/main'
-              name: Coveralls
+    coveralls:
+        runs-on: ubuntu-latest
+        if: needs.changed-files.outputs.any_changed == 'true' && github.event_name == 'push' && github.ref == 'refs/heads/main'
```

Comment:
@cedoor 
`github.event_name == 'push' && github.ref == 'refs/heads/main'` means coveralls will only run on push events to `main` right?

Reply:
Yes!